src/DeviceInterface/InterfacePrototype.js

```diff
@@ -262,6 +263,24 @@ class InterfacePrototype {
         return this.#data.creditCards;
     }
 
+    /**
+     * @param {Map<HTMLElement, import("../Form/Form").Form>} forms
+     */
+    initGlobalFocusHandler(forms) {
+        window.addEventListener(
+            'focus',
+            (e) => {
+                const isAnyFormAutofilling = [...forms.values()].some((form) => form.isAutofilling);
+                const form = [...forms.values()].find((form) => form.hasFocus());
+                const targetElement = pierceShadowTree(e);
+                if (!isAnyFormAutofilling && this.globalConfig.isIOS && targetElement && !(targetElement instanceof Window)) {
+                    this.attachKeyboard({ device: this, form, element: targetElement });
+                }
+            },
+            true,
+        );
```

Comment:
This should likely move to a separate file in the future. I can see how we'll be adding more logic to the focus event. But for now, I'm ok with leaving it here.